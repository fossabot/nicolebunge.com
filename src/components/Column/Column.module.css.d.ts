declare const styles: {
  readonly columnSpan1: string;
  readonly columnSpan2: string;
  readonly columnSpan3: string;
  readonly columnSpan4: string;
  readonly columnSpan5: string;
  readonly columnSpan6: string;
  readonly columnSpan7: string;
  readonly columnSpan8: string;
  readonly columnSpan9: string;
  readonly columnSpan10: string;
  readonly columnSpan11: string;
  readonly columnSpan12: string;
  readonly columnStart1: string;
  readonly columnStart2: string;
  readonly columnStart3: string;
  readonly columnStart4: string;
  readonly columnStart5: string;
  readonly columnStart6: string;
  readonly columnStart7: string;
  readonly columnStart8: string;
  readonly columnStart9: string;
  readonly columnStart10: string;
  readonly columnStart11: string;
  readonly columnStart12: string;
};
export = styles;
