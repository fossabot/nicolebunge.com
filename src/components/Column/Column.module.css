.column--span-1 {
  grid-column-end: span 1;
}

.column--span-2 {
  grid-column-end: span 2;
}

.column--span-3 {
  grid-column-end: span 3;
}

.column--span-4 {
  grid-column-end: span 4;
}

.column--span-5 {
  grid-column-end: span 5;
}

.column--span-6 {
  grid-column-end: span 6;
}

.column--span-7 {
  grid-column-end: span 7;
}

.column--span-8 {
  grid-column-end: span 8;
}

.column--span-9 {
  grid-column-end: span 9;
}

.column--span-10 {
  grid-column-end: span 10;
}

.column--span-11 {
  grid-column-end: span 11;
}

.column--span-12 {
  grid-column-end: span 12;
}

.column--start-1 {
  grid-column-start: 1;
}

.column--start-2 {
  grid-column-start: 2;
}

.column--start-3 {
  grid-column-start: 3;
}

.column--start-4 {
  grid-column-start: 4;
}

.column--start-5 {
  grid-column-start: 5;
}

.column--start-6 {
  grid-column-start: 6;
}

.column--start-7 {
  grid-column-start: 7;
}

.column--start-8 {
  grid-column-start: 8;
}

.column--start-9 {
  grid-column-start: 9;
}

.column--start-10 {
  grid-column-start: 10;
}

.column--start-11 {
  grid-column-start: 11;
}

.column--start-12 {
  grid-column-start: 12;
}
