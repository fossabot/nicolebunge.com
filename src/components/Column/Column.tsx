import classNames from 'classnames';
import React from 'react';
import Alignment, { AlignmentProps } from '../Alignment';
import * as styles from './Column.module.css';

type Span = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;
type Start = 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 11 | 12;

type SpanClass = `columnSpan${Span}`;
type StartClass = `columnStart${Start}`;

export interface ColumnProps extends AlignmentProps {
  span?: Span;
  start?: Start;
}

function Column(props: ColumnProps): JSX.Element {
  const { className, span, start, ...otherProps } = props;

  return (
    <Alignment
      className={classNames(className, {
        [styles[`columnSpan${span}` as SpanClass]]: span,
        [styles[`columnStart${start}` as StartClass]]: start,
      })}
      {...otherProps}
    />
  );
}

export default Column;
