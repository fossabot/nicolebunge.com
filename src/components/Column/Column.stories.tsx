import Column from '.';

export default {
  component: Column,
};

export const Default = {
  args: {
    children: 'Column',
  },
};
