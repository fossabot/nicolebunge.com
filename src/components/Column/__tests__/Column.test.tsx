/* eslint-disable import/no-extraneous-dependencies */

import { render } from '@testing-library/react';
import React from 'react';
import Column from '..';

describe('Column component', () => {
  it('renders correctly', () => {
    const { container } = render(<Column>Column</Column>);

    expect(container.firstChild).toMatchSnapshot();
  });
});
