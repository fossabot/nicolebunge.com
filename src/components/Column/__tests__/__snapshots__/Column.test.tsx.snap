// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Column component renders correctly 1`] = `
<div
  class="columnSpan1 columnStart1"
>
  Column
</div>
`;
